contracts/settling_game/modules/food/Food.cairo

```diff
@@ -0,0 +1,328 @@
+from contracts.settling_game.interfaces.imodules import IL04_Calculator
+
+# -----------------------------------
+# Events
+# -----------------------------------
+
+# @storage_var
+# func Food(token_id : felt) -> (farms_left : felt, last_farm_update : felt):
+# end
+
+# -----------------------------------
+# Storage
+# -----------------------------------
+
+# -------FARMS-------#
+@storage_var
+func farms_built(token_id : Uint256) -> (farms_built : felt):
+end
+
+# each farms build can be harvested X times
+@storage_var
+func farm_harvests_left(token_id : Uint256) -> (farm_harvests_left : felt):
+end
+
+@storage_var
+func last_farm_update(token_id : Uint256) -> (last_farm_update : felt):
+end
+
+# -------FISHING-------#
+@storage_var
+func fishing_villages_built(token_id : Uint256) -> (fishing_villages_built : felt):
+end
+
+# each farms build can be harvested X times
+@storage_var
+func fishing_villages_harvests_left(token_id : Uint256) -> (fishing_villages_harvests_left : felt):
+end
+
+@storage_var
+func last_fishing_villages_update(token_id : Uint256) -> (last_fishing_villages_update : felt):
+end
+
+# -------STORE HOUSE-------#
+@storage_var
+func store_house(token_id : Uint256) -> (available_food : felt):
+end
+
+# -----------------------------------
+# INITIALIZER & UPGRADE
+# -----------------------------------
+
+# @notice Module initializer
+# @param address_of_controller: Controller/arbiter address
+# @proxy_admin: Proxy admin address
+@external
+func initializer{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    address_of_controller : felt, proxy_admin : felt
+):
+    Module.initializer(address_of_controller)
+    Proxy.initializer(proxy_admin)
+    return ()
+end
+
+# @notice Set new proxy implementation
+# @dev Can only be set by the arbiter
+# @param new_implementation: New implementation contract address
+@external
+func upgrade{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    new_implementation : felt
+):
+    Proxy.assert_only_admin()
+    Proxy._set_implementation_hash(new_implementation)
+    return ()
+end
+
+# -----------------------------------
+# EXTERNAL
+# -----------------------------------
+
+@external
+func create{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    token_id : Uint256, qty : felt, food_building_id : felt
+):
+    alloc_locals
+    # check id
+    Food.assert_ids(food_building_id)
+
+    # checks
+    Module.ERC721_owner_check(token_id, ExternalContractIds.S_Realms)
+
+    let (realms_address) = Module.get_external_contract_address(ExternalContractIds.Realms)
+    let (realm_data) = realms_IERC721.fetch_realm_data(realms_address, token_id)
+
+    # checks
+    if food_building_id == RealmBuildingsIds.Farm:
+        let (enough_rivers) = is_le(qty, realm_data.rivers + 1)
+        with_attr error_message("FOOD: Not enough Rivers"):
+            assert_not_zero(enough_rivers)
```

Comment:
`assert enough_rivers = TRUE` is a bit more optimized and IMO more readable as well. 

Applies to the `else` branch below as well.